numpy/core/src/multiarray/descriptor.c

```diff
@@ -1003,48 +1003,34 @@ _use_fields_dict(PyObject *obj, int align)
 static PyArray_Descr *
 _convert_from_dict(PyObject *obj, int align)
 {
-    PyArray_Descr *new;
-    PyObject *fields = NULL;
-    PyObject *names  = NULL;
-    PyObject *offsets= NULL;
-    PyObject *descrs = NULL;
-    PyObject *titles = NULL;
```

Comment:
I do not trust this one. The `goto fail` includes `XDECREF` of these variables.  I do not think they are guaranteed to be initialized to NULL when set directly.

Reply:
Was hoping compiler warnings would catch this type of thing :(

Reply:
Hehe, I think clang-tidy probably does (although I did not pull to see).

Reply:
OTOH, maybe the goto control flow is too complex for it, who knows.

Reply:
Again, this one is safe - we don't `goto fail` until after the variables are defined.

I think if we did, we would get a compiler warning: https://godbolt.org/z/UahE4F

Reply:
Right, sorry. Had not expected that they can be initizialized to NULL there, but of course they can.

Reply:
They're reset to null further down once the ownership is transferred out of them